File: src/api/SkyRoute/src/SkyRoute.BusinessLogic/Domain/Booking.cs
namespace SkyRoute.BusinessLogic.Domain;

public sealed record PassengerDetails(
    string FullName,
    string Email,
    string DocumentNumber);

public sealed record BookingRequest(
    string ProviderKey,
    string FlightNumber,
    string OriginCode,
    string DestinationCode,
    DateTimeOffset DepartureUtc,
    DateTimeOffset ArrivalUtc,
    CabinClass Cabin,
    int Passengers,
    decimal PricePerPassenger,
    string Currency,
    IReadOnlyList<PassengerDetails> PassengerDetails);

public sealed record BookingConfirmation(
    string Reference,
    decimal TotalPrice,
    string Currency,
    DateTimeOffset CreatedAtUtc);

File: src/web/sky-route/src/app/shared/models/booking.ts
import { CabinClass } from './cabin-class';

export interface PassengerDetails {
  fullName: string;
  email: string;
  documentNumber: string;
}

export interface BookingRequest {
  providerKey: string;
  flightNumber: string;
  originCode: string;
  destinationCode: string;
  departureUtc: string;
  arrivalUtc: string;
  cabin: CabinClass | number;
  passengers: number;
  pricePerPassenger: number;
  currency: string;
  passengerDetails: PassengerDetails[];
}

export interface BookingConfirmation {
  reference: string;
  totalPrice: number;
  currency: string;
  createdAtUtc: string;
}
